src/components/CampaignForm.tsx

```diff
@@ -0,0 +1,204 @@
+
+    const [searchFieldValue, setSearchFieldValue] = useState<string>('');
+    const { isIdle, refetch, data: results } = useQuery(
+        ['searchDrawerResults', searchFieldValue],
+        getPeopleSearchResults(searchFieldValue, orgId as string),
+        { enabled: false },
+    );
+
+    const debouncedQuery = useDebounce(async () => {
+        refetch();
+    }, 600);
+
+    // Watch for changes on the search field value and debounce search if changed
+    useEffect(() => {
+        if (searchFieldValue.length >= 3)
+            setSearchLabel(intl.formatMessage({ id: 'misc.formDialog.campaign.searching' }));
+        debouncedQuery();
+        if (results) {
+            setPeople(results);
+        }
+    }, [searchFieldValue, debouncedQuery, results, intl]);
+
+    const initialValues = {
+        info_text: campaign?.info_text,
+        status: campaign?.published,
+        title: campaign?.title,
+        visibility: campaign?.visibility,
+    };
+
+    const validate = (values: Record<string, string>) => {
+        const errors:Record<string, string> = {};
+        if (!values.title) {
+            errors.title = intl.formatMessage({ id: 'misc.formDialog.required' });
+        }
+        return errors;
+    };
+
+    const formFields = [
+        {
+            field: (
+                <TextField
+                    fullWidth
+                    id="title"
+                    label={ intl.formatMessage({ id: 'misc.formDialog.campaign.name' }) }
+                    margin="normal"
+                    name="title"
+                    required
+                />
+            ),
+            size: 12,
+        },
+        {
+            field: (
+                <TextField
+                    fullWidth id="info_text"
+                    label={ intl.formatMessage({ id: 'misc.formDialog.campaign.description' }) }
+                    margin="normal"
+                    multiline
+                    name="info_text"
+                    rows={ 5 }
+                    variant="outlined"
+                />
+            ),
+            size: 12,
+        },
+        {
+            field: (
+                <Autocomplete
+                    defaultValue={{ first_name:campaign?.manager?.name.split(' ')[0], id: campaign?.manager?.id, last_name: campaign?.manager?.name.split(' ')[1] } as Partial<ZetkinPerson>  || null}
+                    getOptionLabel={ person => `${person.first_name || ''} ${person.last_name || ''}` }
+                    getOptionValue={ person => person.id || 0 }
+                    label={ intl.formatMessage({ id: 'misc.formDialog.campaign.manager' }) }
+                    name="manager_id"
+                    noOptionsText={
+                        (!people.length && !isIdle) ? intl.formatMessage({ id: 'misc.formDialog.campaign.noResult' })
+                            : searchLabel }
```

Comment:
It should be possible to derive this from `isLoading`, `results.length` and/or `searchFieldValue.length`, either inline, using a helper function, or by storing the correct label in a temporary variable in this component (not component state, just an ordinary variable) similarly to this, but with i18n:

```ts
let searchLabel = searchFieldValue.length? 'Keep typing to search' : 'Start typing to search';

if (isLoading) {
  searchLabel = 'Searching...';
}
else if (results?.length == 0) {
  searchLabel = 'No options';
}
```

Reply:
nice, for some reason i assumed that without setting the label in state, it wouldn't re-render, but this works better and is much more clear

Reply:
and ofcourse usequery does exactly that, you are able to use the status to render conditionally 0_0